src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11200,28 +11198,49 @@ namespace UnitarySystems {
 
                 DXCoils::SimDXCoil(
                     state, blankString, CompOn, FirstHVACIteration, CompIndex, this->m_FanOpMode, PartLoadRatio, OnOffAirFlowRatio, CoilCoolHeatRat);
-                this->m_CoolCompPartLoadRatio = PartLoadRatio * double(CompOn);
+                if (CompOn == DataHVACGlobals::CompressorOperation::On) {
+                    this->m_CoolCompPartLoadRatio = PartLoadRatio;
+                } else {
+                    this->m_CoolCompPartLoadRatio = 0;
+                }
```

Comment:
Unrolled these multiplying with `double(CompOn)` to if else statements checking if the Compressor is on or of.

Reply:
A shorthand for this idiom is:

```cpp
this->m_CoolCompPartLoadRatio = DataHVACGlobals::CompressorOperation::On ? PartLoadRatio : 0.0;
```

Reply:
Um ... 

```cpp
this->m_CoolCompPartLoadRatio = CompOn ? PartLoadRatio : 0.0;
```

Reply:
You caught me!

Reply:
By the way, I made that mistake on purpose to see if anyone was paying attention! 😉

Reply:
Will make the changes, thanks for the suggestion! 😃

Reply:
#9052 should easily merge with all these changes ...